lib/phase/flatten.js

```diff
@@ -3,6 +3,37 @@
 const Hoek = require('hoek');
 const clone = require('clone');
 
+/**
+ * Check if a job has duplicate steps
+ * @method checkAdditionalRules
+ * @param  {Object}             doc Document that went through structural parsing
+ * @return {Array}                  List of errors
```

Comment:
`or null if no error` should be added to function comment because `[]` in JavaScript is true-ish value. We easily understand the behavior if there is a comment like that.
> if(result)
or 
> if(result.length)

Reply:
fixed